src/import/YAMLConfiguration.ts

```diff
@@ -0,0 +1,380 @@
+  // The parameters property represents IG.definition.parameter. Rather than a list of code/value
+  // pairs (as in the ImplementationGuide resource, the code is the YAML key. If a parameter allows
+  // repeating values, the value in the YAML should be a sequence/array. For a partial list of
+  // allowed parameters see: https://confluence.hl7.org/display/FHIR/Implementation+Guide+Parameters
+  parameters?: YAMLConfigurationParameterMap;
+
+  // The templates property corresponds 1:1 with IG.definition.template. The templates value can be
+  // a single item or a list. Note that the pluralized name 'templates' refers to the
+  // IG.definiton.template definitions; the singularized name 'template' refers to the specific
+  // template to use for this IG.
+  templates?: ImplementationGuideDefinitionTemplate | ImplementationGuideDefinitionTemplate[];
+
+  // The template property will be copied into the ig.ini file. If the value of template is "none",
+  // then only the resources will be generated (with no supporting ImplementationGuide sources).
+  template: string;
+
+  // The following two lines correspond to items that used to be in ig.ini but were recently moved
+  // to IG.definition.parameter. For consistency within this file, the names are represented using
+  // camelcase, but if authors use the formal parameter names, SUSHI will recognize them as well.
+  // In either case, they'll be copied to the IG JSON using the formal names.
+  copyrightYear?: string | number; // YAML will parse years as numbers
+  copyrightyear?: string | number; // YAML will parse years as numbers
+  releaseLabel?: string;
+  releaselabel?: string;
+
+  // The menu property will be used to generate the input/menu.xml file. The menu is represented as
+  // a simple structure where the YAML key is the menu item name and the value is the URL. The IG
+  // publisher currently only supports one level deep on sub-menus.
+  // TO CONSIDER: If no menu data is provided, can we generate the menu based on the pages order
+  // or should we just generate a very standard menu (since there may be too many pages to fit in
+  // a menu)?
+  menu?: YAMLConfigurationMenuTree;
+
+  // The history property corresponds to package-list.json. SUSHI will use the existing top-level
+  // properties in its config to populate the top-level package-list.json properties: package-id,
+  // canonical, title, and introduction. Authors that wish to provide different values can supply
+  // them as properties under history. All other properties under history are assumed to be
+  // versions.
+  //
+  // The current version is special. If the author provides only a single string value, it is
+  // assumed to be the URL path to the current build. The following default values will then be
+  // used:
+  // * desc: Continuous Integration Build (latest in version control)
+  // * status: ci-build
+  // * current: true
+  history?: YAMLConfigurationHistory;
+};
+
+export type YAMLConfigurationMeta = {
```

Comment:
I had to make a shadow YAMLConfiguration type for every type that I needed to change any property for.  For example, I needed to make `YAMLConfigurationMeta` because its `security` and `tag` properties need to also support `string` type since we support FSH-syntax strings...